src/myPyllant/api.py

```diff
@@ -1302,3 +1303,31 @@ async def set_ambisense_room_manual_mode_setpoint_temperature(
         )
         room.room_configuration.temperature_setpoint = temperature
         return room
+
+    async def set_ambisense_room_time_program(
+        self, room: AmbisenseRoom, time_program: RoomTimeProgram
+    ) -> AmbisenseRoom:
+        """
+        Set time program for an ambisense room
+
+        Parameters:
+            room: The target room
+            time_program: The new time program
+        """
+        url = f"{await self.get_api_base()}/api/v1/ambisense/facilities/{room.system_id}/rooms/{room.room_index}/timeprogram"
+
+        data = asdict(time_program)
+        del data["extra_fields"]
+        del data["meta_info"]
+        for day in data.values():
```

Comment:
Maybe there is a more pythonic way to do it ?

Reply:
Gave it a shot here https://github.com/signalkraft/myPyllant/pull/81/commits/882ed8554ac2b449e820d8322e1245ca1bab153d - I added a testcase too, to make sure it results in the same output as your code.